Makefile

```diff
@@ -0,0 +1,106 @@
+# Copyright 2017, 2019 the Velero contributors.
```

Comment:
I'm fine merging this more or less as-is for now, but I do think we can significantly simplify later - we can look at a multi-stage docker build, a basic `go build ...`-based target for building a binary locally, and one or two others for CI & image-pushing.